src/Processors/Transforms/AggregatingInOrderTransform.cpp

```diff
@@ -143,6 +143,7 @@ void AggregatingInOrderTransform::consume(Chunk chunk)
                     source_column = source_column->cut(key_begin, rows - key_begin);
 
                 current_chunk = Chunk(source_columns, rows - key_begin);
+                src_rows -= current_chunk.getNumRows();
```

Comment:
Later we're going to call `consume` again with this new Chunk as argument, right? I spent ten minutes already trying to understand this control flow, but without much success :)

Reply:
Correct (I was confused at first too)